src/Vault.sol

```diff
@@ -458,7 +452,7 @@ contract Vault is IVault, ERC20PermitUpgradeable, AccessControlUpgradeable, Reen
         returns (uint256, uint256)
     {
         uint256 baseAssets = _convertAssetToBase(asset_, assets);
-        uint256 shares = baseAssets.mulDiv(totalAssets() + 1, totalSupply() + 10 ** 0, rounding);
+        uint256 shares = baseAssets.mulDiv(totalSupply() + 10 ** 0, totalAssets() + 1, rounding);
```

Comment:
this fix looks correct as in it should be 

BaseAssets * totalSupply / totalAssets and decimal offset is hardcoded to zero here